db/db_impl/db_impl_write.cc

```diff
@@ -1883,9 +1895,10 @@ Status DBImpl::TrimMemtableHistory(WriteContext* context) {
 }
 
 Status DBImpl::ScheduleFlushes(WriteContext* context) {
-  assert(immutable_db_options_.flush_switch->IsFlushOn());
   autovector<ColumnFamilyData*> cfds;
+  ColumnFamilyData* cfd_flush_disabled = nullptr;
```

Comment:
Same as above, you don't need any changes to `DBImpl::ScheduleFlushes`

Reply:
`ScheduleFlushes` will be triggered when single memtable size exceed the limit. So it's actually automatic flush right?

Reply:
AFAIK `ScheduleFlushes()` drains the queue of flushes, but with `disable_auto_flush` the queue will be empty. Also `ScheduleFlushes()` handles both manual and automatic flushes I think.

Reply:
ahh I see your point. yes. This should never be triggered as long as `disable_auto_flush=true` and no manual flush.

Reply:
Actually we have to do some checks at `ScheduleFlushes` for atomic_flush. Enabling auto flush for multiple CFs is not atomic. So following scenario is possible:

```
1. CF1 has flush enabled and it's put into `flush_scheduler_`
2. CF2 still has flush disabled
```
When we call `ScheduleFlushes` and `atomic_flush=true`, we should return immediately (there is one added test for this case).

Reply:
One thing worth mentioning is, for a follower that tries to take over, it's possible to have all the CFs getting flushed again right after we call `EnableAutoFlush`, even if the memtables have been recently switched. For example,

assuming we have two CFs, for a follower which tries to take over
```
1. CF1->EnableAutoFlush()
2. CF1 is put into `flush_scheduler_` to flush(notice that CF2 hasn't enabled auto flush yet, so `ScheduleFlushes` won't actually clear the `flush_scheduler_`)
3. follower receives `kMemtableSwitch` from leader, so all memtables switched
4. CF2->EnableAutoFlush()
5. `ScheduleFlushes` is called since `flush_scheduler_` is not empty (it has CF1 in it), then another `kMemtableSwitch` is generated
```
So we might see one `kMemtableSwitch` event follows another `kMemtableSwitch` in the log. This is weird but it shouldn't bring any correctness issue. Also we should write some tests to verify in rockset.

Reply:
> 

Great thinking.